<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Biologist's bioinformatics notes</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <script src="/js/jquery.min.js"></script>
  <script src="/js/bootstrap.min.js"></script>
  <link href="/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="/css/theme.css" rel="stylesheet" type="text/css">
  <link href="/css/syntax.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <meta name="google-site-verification" content="fhL4tu7YrFAnIGAUl7DqfxgyLuyY6vE5h8o6fUwzoJM" />
</head>

<body>
  <nav class="navbar-fixed-top bg-primary">
    <span class="navbar-brand">Bioinformatics notes from a biologist</span>
    <ul class="nav navbar-nav">
      <li class="active"><a href="/"><i class="fas fa-home fa-2x"></i> Home</a></li>
      <li class="active"><a href="/archive.html"><i class="fas fa-archive fa-2x"></i> Archive</a></li>
    </ul>
  </nav>
</body>


<div class="container-fluid">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class=well>
  <div class="sidebar">
    28 May 2021
  </div>
  <div class="sidebar">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="/gc_calculation_python">Calculate GC content from multiline fasta with gaps in sequences</a></li>
        
          <li><a href="/split_snp_records_to_fa">Split each sequence with two alleles into two sequences and convert the text to fasta</a></li>
        
          <li><a href="/fill_na_values">Replace empty values in text files with awk</a></li>
        
          <li><a href="/percentage_bars">Bar plot with unbalanced data</a></li>
        
          <li><a href="/trim_sequences">Trim fasta sequences based on a sequence in fasta file</a></li>
        
    </ul>
  </div>
  <div class="sidebar">
    <h1>Links</h1>
<ul>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/svsk/">Linkedin</a></li>
  <li><a target="_blank" rel="noopener noreferrer" href="https://www.github.com/svsuresh">Github</a></li>
</ul>

  </div>
</div>
        </div>
        <div class="col-md-9">
          <div class="article">
            <div class="well">
                <h1><a href="/generate-random-dna-elements-using">Jun 8, 2016 - Generate random DNA elements using python/biopython</a></h1>
            <div class="post-content">
            <div dir="ltr" style="text-align: left;" trbidi="on">
Sequence simulations are useful in several studies in bioinformatics. Such simulations are covered <a href="http://digibio.blogspot.com/2015/07/generate-random-nucleotide-and-amino.html" target="_blank">here</a> in earlier blogs. Note that the simulation scripts are written in R. In this note, we will use python/biopython to simulate DNA elements. Objective of the study is to simulate DNA sequence after taking following inputs from user:<br />
1) Length of the desired sequence<br />
2) Desired number of sequences<br />
3) IUPAC code or Standard code (ATGC)<br />
<br />
script will take above input and stores the sequences in the same folder where code is executed with date and time stamp. Stored format is fasta.<br />
<br />
============================================<br />
from Bio.Alphabet import IUPAC<br />import random<br />from datetime import datetime<br /><br />n=int(input("type the length of the sequence: "))<br />j=int(input("type the number of sequences: "))<br />k=(input("choose S for standard code and I for IUPAC: "))<br /><br />if k == "S":<br />&nbsp;&nbsp;&nbsp; sequnce=IUPAC.IUPACUnambiguousDNA.letters<br />elif k == "I":<br />&nbsp;&nbsp;&nbsp; sequnce=IUPAC.IUPACAmbiguousDNA.letters<br />dnafile = open("dnasequence_"+datetime.now().strftime("%Y%m%d_%H%M%S")+".fa", "a")<br />for j in range(j):<br />&nbsp;&nbsp;&nbsp; my_seq=''.join(random.choice(sequnce) for i in range(n))<br />&nbsp;&nbsp;&nbsp; id="seq "+str(j+1)<br />&nbsp;&nbsp;&nbsp; my_seq = "&gt;"+id+"\n"+my_seq <br />&nbsp;&nbsp;&nbsp; dnafile.write(my_seq+"\n")<br />dnafile.close()<br />
============================================<br />
<br />
Please note that python is particular about indentation and make sure that indentation is correct. </div>

            </div>
          </div>
          <div class="pagination">
              
                <a class="btn btn-default" href="/simulate-protein-sequences" class="next">Newer Post</a>
              
              
                <a class="btn btn-default" href="/plotting-variants-per-chromosome-using" class="previous">Older Post</a>
              
          </div>
        </div>
    </div>
</div>

<body>
<html>	
<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2021 Biologist's bioinformatics notes. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>

</body>
</html>

